### PR TITLE
Enable CI, move some functions to a separate file

### PR DESCRIPTION
**Pull request by mkaito**
I stole some code from `serokell/profiles` to build any derivations in the overlay, plus a check for trailing whitespace.

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+let
+  overlay = import ./pkgs;
+  nixpkgs = import (import ./nixpkgs.nix) {
+    config.allowUnfree = true;
```
Comment (Lucus16) [inline release.nix]: I don't think it's a good idea to set `allowUnfree` in an open source project.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/result*
```
Comment (Lucus16) [inline .gitignore]: newline at end of file plz

**Review by Lucus16 — CHANGES_REQUESTED** [review]

File: release.nix (hunk 1 above)
Comment (mkaito) [reply release.nix]: Depends on what you put in the overlay, I guess.

Comment (Lucus16) [reply release.nix]: No, the point is that you make that decision for everything in the nixpkgs pin and this option exists in order to make that an explicit choice. Even if the whole closure was useless without `allowUnfree = true`, users should still be required to pass that explicitly in order to acknowledge they're willing to use unfree software.

**Review by yorickvP — APPROVED** [review]
the nginx patch is upstream already. gitignore should be pointed to hercules/gitignore (or upstream if it's upstreamed). We can just about get rid of this repo, maybe. But enabling Ci in the meantime is good.

**Pull request merged**